images/operator/Dockerfile

```diff
@@ -2,6 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ARG BASE_IMAGE=scratch
+ARG CILIUM_BUILDER_IMAGE=quay.io/cilium/cilium-builder:f55691636922518ca40802f59d57f36ee4741c0d@sha256:4c21a3e21e9e07f0301191b3ae6921c8208a3910ce1b73be6ad445e544590e3a
```

Comment:
This image sha differs from the one in `images/cilium/Dockerfile`. Hmm. Usually if we change the cilium builder or runtime images, we need to push the PR to the cilium/cilium repo directly and then there's a bot that will autobuild the runtime/builder images and update these SHAs. I suspect that if we merged this PR as-is, something may break but I'm not entirely sure.

@aanm what do you think the best workflow will be here? Strictly speaking the new code doesn't need to be in the builder image, but it won't change often so it's kinda nice to build it once generically so that it can be reused in both the operator and cilium images.

Reply:
@joestringer the `make -C images/ update-builder-image` will pick these changes automatically. No need to change anything in this PR.

Reply:
OK, so you're saying we can just ignore the failing "[Base Image Lint](https://github.com/cilium/cilium/actions/runs/4251959190/jobs/7394959081)" job below, merge this once it's passed review, then immediately open a PR to update the builder image using the above?

Reply:
Or are you saying that once [this build](https://github.com/cilium/cilium/actions/runs/4251959003) completes, Marco can just run the make command above to pull in the right image version?

Reply:
rebased to master to pull in the latest runtime/builder images which seems to be changed on master. this should fix the base image lint issue and hopefully let me trigger the builder image build with my changes :pray:

Reply:
cilium-builder image built successfully

pushed a commit which changes the usages to the corresponding image